cub/device/dispatch/dispatch_batch_memcpy.cuh

```diff
@@ -0,0 +1,631 @@
+    if (CubDebug(error = MaxSmOccupancy(batch_memcpy_blev_occupancy, multi_block_memcpy_kernel, BLEV_BLOCK_THREADS)))
+      return error;
+
+    uint32_t batch_memcpy_blev_grid_size = sm_count * batch_memcpy_blev_occupancy;
+
+    // Construct the tile status for the buffer prefix sum
+    BLevBufferOffsetTileState buffer_scan_tile_state;
+    if (CubDebug(error = buffer_scan_tile_state.Init(num_tiles,
+                                                     allocations[MEM_BLEV_BUFFER_SCAN_STATE],
+                                                     allocation_sizes[MEM_BLEV_BUFFER_SCAN_STATE])))
+    {
+      return error;
+    }
+
+    // Construct the tile status for thread blocks-to-buffer-assignment prefix sum
+    BLevBlockOffsetTileState block_scan_tile_state;
+    if (CubDebug(error = block_scan_tile_state.Init(num_tiles,
+                                                    allocations[MEM_BLEV_BLOCK_SCAN_STATE],
+                                                    allocation_sizes[MEM_BLEV_BLOCK_SCAN_STATE])))
+    {
+      return error;
+    }
+
+    // Invoke init_kernel to initialize buffer prefix sum-tile descriptors
+    THRUST_NS_QUALIFIER::cuda_cub::launcher::triple_chevron(init_grid_size, INIT_KERNEL_THREADS, 0, stream)
+      .doit(init_buffer_scan_states_kernel, buffer_scan_tile_state, num_tiles);
+
+    // Check for failure to launch
+    if (CubDebug(error = cudaPeekAtLastError()))
+      return error;
+
+    // Sync the stream if specified to flush runtime errors
+    if (debug_synchronous && (CubDebug(error = SyncStream(stream))))
+      return error;
+
+    // Invoke init_kernel to initialize thread blocks-to-buffer-assignment prefix sum-tile descriptors
+    THRUST_NS_QUALIFIER::cuda_cub::launcher::triple_chevron(scan_grid_size, INIT_KERNEL_THREADS, 0, stream)
+      .doit(init_block_scan_states_kernel, block_scan_tile_state, num_tiles);
+
+    // Check for failure to launch
+    if (CubDebug(error = cudaPeekAtLastError()))
+      return error;
+
+    // Sync the stream if specified to flush runtime errors
+    if (debug_synchronous && (CubDebug(error = SyncStream(stream))))
+      return error;
+
+    // Invoke kernel to copy small buffers and put the larger ones into a queue that will get picked up by next kernel
+    THRUST_NS_QUALIFIER::cuda_cub::launcher::triple_chevron(batch_memcpy_grid_size,
+                                                            ActivePolicyT::AgentSmallBufferPolicyT::BLOCK_THREADS,
+                                                            0,
+                                                            stream)
+      .doit(bach_memcpy_non_blev_kernel,
+            input_buffer_it,
+            output_buffer_it,
+            buffer_sizes,
+            num_buffers,
+            d_blev_src_buffers,
+            d_blev_dst_buffers,
+            d_blev_buffer_sizes,
+            d_blev_block_offsets,
+            buffer_scan_tile_state,
+            block_scan_tile_state);
+
```

Comment:
Should we check the error after the previous kernel launch?

Reply:
Fixed by checking the return of `doit()`